ExampleMod/Tiles/ExampleHerb.cs

```diff
@@ -7,78 +9,82 @@
-			//TileObjectData.addBaseTile(out TileObjectData.StyleAlch);
+			Main.tileCut[Type] = true;
+			Main.tileNoFail[Type] = true;
+
 			TileObjectData.newTile.CopyFrom(TileObjectData.StyleAlch);
-			TileObjectData.newTile.AnchorValidTiles = new[]
+
+			TileObjectData.newTile.AnchorValidTiles = new int[]
 			{
-				2, //TileID.Grass
-				109, // TileId.HallowedGrass
+				TileID.Grass,
+				TileID.HallowedGrass,
 				TileType<ExampleBlock>()
 			};
-			TileObjectData.newTile.AnchorAlternateTiles = new[]
+
+			TileObjectData.newTile.AnchorAlternateTiles = new int[]
 			{
-				78, //ClayPot
+				TileID.ClayPot,
 				TileID.PlanterBox
 			};
-			TileObjectData.addTile(Type);
-			//drop = ItemType()
-		}
-		//public override bool CanPlace(int i, int j)
-		//{
-		//	return base.CanPlace(i, j);
-		//}
-		public override void SetSpriteEffects(int i, int j, ref SpriteEffects spriteEffects) {
-			if (i % 2 == 1) {
+
+			TileObjectData.addTile(Type);
+		}
+
+		public override void SetSpriteEffects(int i, int j, ref SpriteEffects spriteEffects)
+		{
+			if (i % 2 == 1)
 				spriteEffects = SpriteEffects.FlipHorizontally;
-			}
 		}
 
-		public override bool Drop(int i, int j) {
-			int stage = Main.tile[i, j].frameX / 18;
-			if (stage == 2) {
-				Item.NewItem(i * 16, j * 16, 0, 0, ItemType<Items.Placeable.ExampleHerbSeeds>());
-			}
+		public override bool Drop(int i, int j)
+		{
+			Stage stage = GetStage(i, j); //The current stage of the herb
+
+			//Only drop items if the herb is grown
+			if (stage == Stage.Grown)
+				Item.NewItem(new Vector2(i, j).ToWorldCoordinates(), ItemType<ExampleHerbSeeds>());
+
 			return false;
 		}
 
-		public override void RandomUpdate(int i, int j) {
-			if (Main.tile[i, j].frameX == 0) {
-				Main.tile[i, j].frameX += 18;
-			}
-			else if (Main.tile[i, j].frameX == 18) {
-				Main.tile[i, j].frameX += 18;
-			}
-		}
-		//public override void RightClick(int i, int j)
-		//{
-		//	base.RightClick(i, j);
-		//}
+		public override void RandomUpdate(int i, int j)
+		{
+			Tile tile = Framing.GetTileSafely(i, j); //Safe way of getting a tile instance
+			Stage stage = GetStage(i, j); //The current stage of the herb
+
+			//Only grow to the next stage if there is a next stage. We dont want our tile turning pink!
+			if (stage != Stage.Grown) {
+				//Increase the x frame to change the stage
+				tile.frameX += pixelsPerStage;
+
+				//If in multiplayer, sync the frame change
+				if (Main.netMode != NetmodeID.SinglePlayer)
+					NetMessage.SendTileSquare(-1, i, j, 1);
+			}
+		}
+
+		//A method to quickly get the current stage of the herb
+		private Stage GetStage(int i, int j)
+		{
+			Tile tile = Framing.GetTileSafely(i, j); //Always use Framing.GetTileSafely instead of Main.tile as it prevents any errors caused from other mods
+			return (Stage)(tile.frameX / pixelsPerStage);
+		}
+
+		//An enum on the 3 stages of herb growth.
+		private enum Stage : byte

```

Comment:
This should be above the const, preferably even outside the class as `PlantStage` when used for multiple plants.